File: benchmarks/oopsla25/subjects/109.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f109(const char* s) {
    char a = s[0];
    char b = s[1];
    char t[3];
    t[0] = a;
    t[1] = b;
    t[2] = '\0';
    assert(strcmp(t, "aa") == 0);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f109(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/109.py
#!/usr/bin/env python3

import sys

def f109(s: str):
  a,b = s
  t = a + b
  assert t == "aa"

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f109(my_string)
